pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1432,9 +1434,7 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                                             if (topicFuture.isCompletedExceptionally()) {
                                                 log.warn("{} future is already completed with failure {}, closing the"
                                                         + " topic", topic, FutureUtil.getException(topicFuture));
-                                                persistentTopic.stopReplProducers().whenComplete((v, exception) -> {
```

Comment:
In original implements, It ensures that `persistentTopic.stopReplProducers()` executes before `topics.remove(...)`, this ensures that "there are no multiple threads change one topic"

After this change, this advantage was deleted.